src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -154,24 +154,38 @@ public static Task<ILanguageServer> CreateAsync(Stream input, Stream output, Tra
                         services.AddSingleton<HtmlFactsService, DefaultHtmlFactsService>();
                     }));
 
-            server.OnShutdown(() =>
-            {
-                TempDirectory.Instance.Dispose();
-                return Unit.Task;
-            });
-
             try
             {
                 var factory = new LoggerFactory();
                 var logger = factory.CreateLogger<RazorLanguageServer>();
                 var assemblyInformationAttribute = typeof(RazorLanguageServer).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
                 logger.LogInformation("Razor Language Server version " + assemblyInformationAttribute.InformationalVersion);
+                factory.Dispose();
             }
             catch
             {
                 // Swallow exceptions from determining assembly information.
             }
 
+            IDisposable shutdownSubscription = null;
```

Comment:
Any reason to breakup the declaration and assignment for this and the `exitSubscription`?

Reply:
C# restrictions 😄